### DISCUSSION TITLE
$refs.modalElement.showModal() doesn't works

### DISCUSSION THREAD

**Discussion by rafli-sss** [alpinejs/alpine#3964] [1. Help] [answered]
```blade
<li x-data="{
  openModal() {
    $refs.logoutModal.showModal();
  },
  closeModal() {
    $refs.logoutModal.close();
  }
}">
  <button
    class="block w-full px-4 py-2 text-start text-sm hover:bg-gray-100 dark:hover:bg-gray-600 dark:hover:text-white"
    type="button"
    x-on:click="openModal()"
  >
    Log out
  </button>

  <dialog
    class="max-w-96 w-1/2 rounded transition backdrop:bg-gray-600/75 backdrop:backdrop-blur-sm"
    id="logout-modal"
    x-ref="logoutModal"
  >
    <div
      class="px-4 py-2"
      x-on:click.outside="closeModal()"
    >
       <div class="border-b pb-8">
         <p class="text-2xl">Are you sure you want to log out?</p>
       </div>

       <div class="">
         <x-button
           type="button"
           variant="secondary"
           x-on:click="closeModal()"
         >
           Cancel
         </x-button>
         <a href="{{ route('logout') }}">Log out</a>
      </div>
    </div>
  </dialog>
</li>
```

Why $refs.logoutModal.showModal() doesn't works?
When i manually open the modal from console (getElementById(...).showModal()) it opens, then click the cancel button and it works (close the modal)

**Comment by ekwoka** [comment]
Can you put `this.$refs` instead? and let me know how that works for you.

**Reply by rafli-sss** [reply]
Still no luck. I did `console.log($refs.logoutModal)` it outputs the `<dialog>` element.

**Comment by SimoTod** [comment, marked as answer]
The dialog opens but then it closes immediately because you also have a click.outside triggering the close action 
Eiither you stop the event on the open button with the stop modifier or you open the modal using $nextTick so the close function triggers first.

**Reply by rafli-sss** [reply]
Thanks, it works!

**Reply by ekwoka** [reply]
right, the `click.outside after click` fix only happens in regards to `click.outside` inside of `x-show`.
